From 36332072971653224b5088e9a86417dc94aaa56f Mon Sep 17 00:00:00 2001
From: dissent1 <be.dissent+github@gmail.com>
Date: Tue, 14 Mar 2017 20:51:09 +0300
Subject: [PATCH] Add independent clocks

---
 drivers/cpufreq/cpufreq-dt-platdev.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/cpufreq/cpufreq-dt-platdev.c b/drivers/cpufreq/cpufreq-dt-platdev.c
index 7126762..0fda518 100644
--- a/drivers/cpufreq/cpufreq-dt-platdev.c
+++ b/drivers/cpufreq/cpufreq-dt-platdev.c
@@ -35,6 +35,8 @@ static const struct of_device_id machines[] __initconst = {
 
 	{ .compatible = "marvell,berlin", },
 
+	{ .compatible = "qcom,ipq8064", },
+
 	{ .compatible = "samsung,exynos3250", },
 	{ .compatible = "samsung,exynos4210", },
 	{ .compatible = "samsung,exynos4212", },
@@ -84,6 +86,10 @@ static const struct of_device_id machines[] __initconst = {
 	{ }
 };
 
+#ifdef CONFIG_KRAITCC
+struct cpufreq_dt_platform_data pdata = { .independent_clocks = true };
+#endif
+
 static int __init cpufreq_dt_platdev_init(void)
 {
 	struct device_node *np = of_find_node_by_path("/");
