From 779fef92e80e03b1e5824e050cc23e358ef8907b Mon Sep 17 00:00:00 2001
From: dissent1 <be.dissent+github@gmail.com>
Date: Tue, 14 Mar 2017 22:04:26 +0300
Subject: [PATCH] Update cpufreq-dt.h

---
 drivers/cpufreq/cpufreq-dt.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/cpufreq/cpufreq-dt.h b/drivers/cpufreq/cpufreq-dt.h
index 54d774e..cbf7a21 100644
--- a/drivers/cpufreq/cpufreq-dt.h
+++ b/drivers/cpufreq/cpufreq-dt.h
@@ -14,6 +14,13 @@
 
 struct cpufreq_dt_platform_data {
 	bool have_governor_per_policy;
+	
+	/*
+	 * True when each CPU has its own clock to control its
+	 * frequency, false when all CPUs are controlled by a single
+	 * clock.
+	 */
+	bool independent_clocks;
 };
 
 #endif /* __CPUFREQ_DT_H__ */
