Index: /src/linux/universal/linux-4.9/include/net/netfilter/nf_conntrack_ecache.h
===================================================================
--- /src/linux/universal/linux-4.9/include/net/netfilter/nf_conntrack_ecache.h	(revision 32626)
+++ /src/linux/universal/linux-4.9/include/net/netfilter/nf_conntrack_ecache.h	(revision 32627)
@@ -94,9 +94,4 @@
 			  u32 portid, int report)
 {
-	const struct net *net = nf_ct_net(ct);
-
-	if (!rcu_access_pointer(net->ct.nf_conntrack_event_cb))
-		return 0;
-
 	return nf_conntrack_eventmask_report(1 << event, ct, portid, report);
 }
@@ -105,9 +100,4 @@
 nf_conntrack_event(enum ip_conntrack_events event, struct nf_conn *ct)
 {
-	const struct net *net = nf_ct_net(ct);
-
-	if (!rcu_access_pointer(net->ct.nf_conntrack_event_cb))
-		return 0;
-
 	return nf_conntrack_eventmask_report(1 << event, ct, 0, 0);
 }
Index: /src/linux/universal/linux-4.9/net/fast-classifier/fast-classifier.c
===================================================================
--- /src/linux/universal/linux-4.9/net/fast-classifier/fast-classifier.c	(revision 32626)
+++ /src/linux/universal/linux-4.9/net/fast-classifier/fast-classifier.c	(revision 32627)
@@ -1384,5 +1384,5 @@
 	if (!test_bit(IPS_FIXED_TIMEOUT_BIT, &ct->status)) {
 		spin_lock_bh(&ct->lock);
-		ct->timeout.expires += sis->delta_jiffies;
+		ct->timeout += sis->delta_jiffies;
 		spin_unlock_bh(&ct->lock);
 	}
Index: /src/linux/universal/linux-4.9/net/shortcut-fe/sfe_cm.c
===================================================================
--- /src/linux/universal/linux-4.9/net/shortcut-fe/sfe_cm.c	(revision 32626)
+++ /src/linux/universal/linux-4.9/net/shortcut-fe/sfe_cm.c	(revision 32627)
@@ -810,5 +810,5 @@
 	if (!test_bit(IPS_FIXED_TIMEOUT_BIT, &ct->status)) {
 		spin_lock_bh(&ct->lock);
-		ct->timeout.expires += sis->delta_jiffies;
+		ct->timeout += sis->delta_jiffies;
 		spin_unlock_bh(&ct->lock);
 	}
