diff --git a/drivers/net/wireless/ath/ath10k/htt_rx.c b/drivers/net/wireless/ath/ath10k/htt_rx.c
index ddd94c53d323..af16c24aeee2 100644
--- a/drivers/net/wireless/ath/ath10k/htt_rx.c
+++ b/drivers/net/wireless/ath/ath10k/htt_rx.c
@@ -1546,10 +1546,8 @@ static int ath10k_htt_rx_handle_amsdu(struct ath10k_htt *htt)
	if (ret < 0) {
		ath10k_warn(ar, "rx ring became corrupted: %d\n", ret);
 		__skb_queue_purge(&amsdu);
-		/* FIXME: It's probably a good idea to reboot the
-		 * device instead of leaving it inoperable.
-		 */
		htt->rx_confused = true;
+		queue_work(ar->workqueue, &ar->restart_work);
 		return ret;
 	}
